R/mirror.R

```diff
@@ -7,11 +6,78 @@ mirror <- function(image_path, direction = 'all') {
   #'It returns an array of pixel values for the mirrored image(s).
   #'
   #' @param image_path string: the file path of the image to be mirrored
-  #' @param direction string: dirction of mirroring. Options: 'horizontal', 'vertical', 'all'. Default setting 'all'
+  #' @param direction string: dirction of mirroring. Options: 'horizontal', 'vertical', 'all'. Defaults to 'all'
   #'
-  #' @return array: pixel values of the mirrored image(s)
+  #' @return array/matrix: pixel values of the original image and mirrored image(s)
   #' @export
   #'
   #' @examples
+  #' mirror("tests/testthat/img/milad.jpg", "horizontal")
+  #' mirror("tests/testthat/img/bw.png", "vertical")
+  #' mirror("tests/testthat/img/bw.png", "all")
+  #' mirror("tests/testthat/img/milad.jpg")
+
+
+  require(OpenImageR)
+
+  # check for correct function inputs
+  if(!is.character(image_path)){
+    stop("Error: image_path is not a string")
+  }ins
+  if(!is.character(direction)){
+    stop("Error: direction is not a string, it must be: 'horizonal', 'vertical', 'all'")
+  }
+  if(!(tolower(direction) %in% c("horizontal","vertical", "all"))){
+    stop("Invalid direction for mirroring. Choose between: 'horizontal', 'vertical', 'all'")
+  }
+  if(!file.exists(image_path)){
+    stop("Incorrect directory/image not found")
+  }
+
+
+  # read in image as array/matrix
+  original <- OpenImageR::readImage(image_path)
+
+  # mirror in horizontal direction
```

Comment:
Would it be possible to use the OpenImageR function flipImage for this task? It might make the code a little more maintainable.